app/controllers/api/v2/puppet_hosts_controller.rb

```diff
@@ -4,23 +4,23 @@ class PuppetHostsController < V2::BaseController
       include Api::Version2
       include HostsControllerExtension
 
+      prepend_before_action :fail_and_inform_about_plugin
+
```

Comment:
Look at this block: https://github.com/theforeman/foreman/pull/7800/files#diff-36882131819ddf97a1e444b073996328R7-R10
It would mark the resource as deprecated in `/apidoc`. I think we can use it here as well.
I'm not sure if it wouldn't deprecate the enpoints added by `HostsControllerExtensions` so it needs to get tested. I'd be ok if we do not have it if that turns out to be too much of strugle. If it works though, it adds visibility to the deprecation.